plugin/transfer/transfer.go

```diff
@@ -0,0 +1,135 @@
+package transfer
+
+import (
+	"context"
+
+	"github.com/coredns/coredns/plugin"
+	clog "github.com/coredns/coredns/plugin/pkg/log"
+	"github.com/coredns/coredns/request"
+
+	"github.com/miekg/dns"
+)
+
+var log = clog.NewWithPlugin(pluginName)
+
+// Transfer is a plugin that handles zone transfers.
+type Transfer struct {
+	xfrs []*xfr
+	Next plugin.Handler
+}
+
+type xfr struct {
+	Zones       []string
+	to          []string
+	Transferers map[string]Transferer // a map of zone->plugins which implement Transferer
+}
+
+// Transferer may be implemented by plugins to enable zone transfers
+type Transferer interface {
+	// AllRecords writes all records in the zone to the channel.  The
+	// first and last records written to the channel must be the SOA for the zone.
+	// If serial is non zero, an incremental zone transfer should be assumed.
+	AllRecords(zone string, serial uint32, ch chan dns.RR)
+
+	// Authoritative must return a list of Zones that the plugin is authoritative for.
+	Authoritative() []string
+}
+
+// ServeDNS implements the plugin.Handler interface.
+func (t Transfer) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
+	state := request.Request{W: w, Req: r}
+
+	var serial uint32
+	if state.QType() == dns.TypeIXFR {
```

Comment:
this needs the ixfr handling that was recently added in file's tranfer

Reply:
I am currently handling this in the plugin implementations, to allow them to implement non-axfr-fallback if they could.  But implementing non-axfr-fallback is probably to much to expect of a plugin, and most plugins probably woudn't bother (the full ixfr answer protocol being a tad fiddly).  So, yeah , makes more practical sense to move the simple ixfr handling to the transfer plugin, and later we can extend the transfer plugin, perhaps with an IncrTransferer interface down the road.

Reply:
Yep, that's was my thinking why `AllRecords` would be a bad name for the functions in the interfaces. Now we can extend this and keep a similar naming scheme